datafusion/optimizer/src/push_down_projection.rs

```diff
@@ -337,7 +338,7 @@ impl OptimizerRule for PushDownProjection {
                     )?);
                     let new_filter = child_plan.with_new_inputs(&[new_projection])?;
 
-                    generate_plan!(projection_is_empty, plan, new_filter)
+                    plan.with_new_inputs(&[new_filter])?
```

Comment:
in this branch projection_is_empty is false so new_filter was returned as is. However that means it becomes a filter plan with a different schema. so this way we keep the original plan and change it's inputs